File: 50-PowxN/50-PowxN.c
// Last updated: 04/04/2026, 15:47:54
double myPow(double x, int n) {
    long long exp = n;   // handle INT_MIN safely

    if (exp < 0) {
        x = 1 / x;
        exp = -exp;
    }

    double result = 1.0;

    while (exp > 0) {
        if (exp & 1)
            result *= x;
        x *= x;
        exp >>= 1;
    }

    return result;
}

File: 50-PowxN/50-PowxN.java
// Last updated: 04/04/2026, 15:47:51
class Solution {
    public double power(double a,long b){
        if(b==1) return a;
        if(b==0) return 1;
        double ans = power(a, b/2);
        if(b%2==0) return ans*ans;
        return ans*ans*a;
    }
    public double myPow(double x, int n) {
        long b = n;  
        if(b < 0){
            x = 1 / x;
            b = -b;
        }
        return power(x,b);
    }
}